lib/localparticipant.js

```diff
@@ -528,20 +507,12 @@ class LocalParticipant extends Participant {
     const trackSignaling = this._signaling.tracks.get(localTrack.id);
     trackSignaling.publishFailed(new Error(`The ${localTrack} was unpublished`));
 
-    const publishedKindTracks = {
-      audio: this.audioTrackPublications,
-      video: this.videoTrackPublications,
-      data: this.dataTrackPublications
-    }[localTrack.kind];
     const localTrackPublication = getTrackPublication(this.trackPublications, localTrack);
-
+    this._removeTrack(localTrack);
     if (localTrackPublication) {
-      publishedKindTracks.delete(localTrackPublication.trackSid);
-      this.trackPublications.delete(localTrackPublication.trackSid);
+      this._removeTrackPublication(localTrackPublication);
```

Comment:
Similar question as above: moved the lines above to `_removeTrackPublication` on Participant? Also, I'm not yet sure about the implications of moving `_removeTrack` to before `_removeTrackPublication`.

Reply:
@markandrus Ideally this will emit `trackUnsubscribed` before `trackUnpublished`.